### PR TITLE
Add button to toggle theme in header

### PR DESCRIPTION
**Pull request by LudvigHz**
The button is an icon that switches based on the current theme. If the
user is logged in, the theme will also be persisted to the
backend.

### CODE REVIEW

File: app/components/Header/ToggleTheme.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      const currentTheme = getTheme();
+      const newTheme = currentTheme === 'dark' ? 'light' : 'dark';
+      applySelectedTheme(newTheme);
+      setIcon(getIcon());
+      loggedIn && username && updateUserTheme(username, newTheme);
+    },
+    [loggedIn, username, updateUserTheme]
+  );
+
+  console.log(icon);
```
Comment (ivarnakken) [inline app/components/Header/ToggleTheme.js:29]: :eyes:

File: app/components/Header/index.js [hunk 2]
```diff
@@ -213,6 +215,12 @@ class Header extends Component<Props, State> {
             </div>
 
             <div className={styles.buttonGroup}>
+              <ToggleTheme
+                loggedIn={loggedIn}
+                updateUserTheme={this.props.updateUserTheme}
+                username={this.props.currentUser?.username}
+              />
```
Comment (ivarnakken) [inline app/components/Header/index.js:222]: How does this look on mobile? The `.buttonGroup` is already very cramped (_note: we should reduce the gap on mobile_). I think I would prefer _not_ to have the toggle in the header on mobile, which sites like [Vite](https://vitejs.dev/) and [Docker](https://docs.docker.com/) do as well.

**Review by ivarnakken — CHANGES_REQUESTED** [review]
This seems to work nicely! 💯 Next out is making the dark theme actually functional 😆

File: app/components/Header/index.js (hunk 2 above)
Comment (eikhr) [reply app/components/Header/index.js:222]: I agree, but I think it should be in the user drop-down or something on mobile, so it's not too hard to get to.

Comment (LudvigHz) [reply app/components/Header/index.js]: I have added it to the dropdown now (when logged in)

**Comment by LudvigHz** [comment]
[Screencast from 08-30-2022 11:25:57 PM.webm](https://user-images.githubusercontent.com/42850232/187546334-92446d8c-c3d5-425a-b5fe-3fa8e0522211.webm)


image
image

**Review by ivarnakken — APPROVED** [review]
Looks awesome now 💯

**Pull request merged**